src/services/user-config-service.ts

```diff
@@ -115,20 +116,31 @@ const convertYamlToUserConfig = (input?: string): Config => {
 
 	const config: Config = YAML.parse(input);
 
-	if (!config?.deployments?.environmentMapping) {
+	const configDeployments = config?.deployments;
+	const deployments = {};
+	if (configDeployments != null) {
+		if (configDeployments.environmentMapping) {
+			deployments["environmentMapping"] =  {
+				development: configDeployments.environmentMapping.development,
+				testing: configDeployments.environmentMapping.testing,
+				staging: configDeployments.environmentMapping.staging,
+				production: configDeployments.environmentMapping.production
+			};
+		}
+		if (configDeployments.services) {
+			deployments["services"] = {
+				ids: configDeployments.services.ids?.slice(0, MAX_SERVICE_ID_COUNT)
```

Comment:
Should we throw an exception if ids is not an array?

Reply:
at common.ts Config we define ids as string[] should we need to check the typeof it here again?